packages/ketcher-react/src/script/editor/tool/AbstractTool.ts

```diff
@@ -0,0 +1,35 @@
+import Editor from '../Editor'
+
+interface ToolEventHandler {
+  click?(event: Event): void
+
+  dblclick?(event: Event): void
+
+  mousedown?(event: Event): void
+
+  mousemove?(event: Event): void
+
+  mouseup?(event: Event): void
+
+  mouseleave?(event: Event): void
+
+  mouseLeaveClientArea?(event: Event): void
+
+  mouseover?(event: Event): void
+}
+
+export interface AbstractTool extends ToolEventHandler {
+  close?(): void
+
+  cancel?(): void
+
+  isSelectionRunning?(): boolean
+
+  isNotActiveTool?: boolean
+}
```

Comment:
I'm wondering why `close()`, `isSelectingRunning()`, `isNotActiveTool()` should be in AbstractTool. I thought that not every tool needed them, right?

Reply:
actually, it `isSelectingRunning()` and `isNotActiveTool` is something that is needed for every tool, as it is an interface used from the outside to control abstract tool.

Example `Editor.ts:633`:
```
  const EditorTool = editor.tool()
  ....
    clientArea.contains(event.target) || EditorTool.isSelectionRunning?.(),
  ]
```

Example `Editor.ts:234`:
```
if (!tool || tool.isNotActiveTool) {
```


Even I think, we should mark those methods as required and implement them in each tool. Consistency leads to simplicity.

Not sure about `.close()` - can't find the case... Looks like it is my mistake.

@yuleicul what do you think about it?

Reply:
Thanks for your examples, make sense